File: rle.py
string = "cccaab"


def rle_encoder(string):
    """
    rle(cccaab) -> c3a2b1
    """
    result = ""
    count = 1
    for i in range(0, len(string)):
        if i == 0:
            result += string[i]
        elif string[i] == string[i - 1]:
            count += 1
        else:
            result += str(count) + string[i]
            count = 1
    result += str(count)
    return result


print(rle_encoder(string))

File: rle.js
let tocode = "cccaab"

//run length encoding string
let encode = (str) => {
    let result = ""
    let count = 1
    for (let i = 0; i < str.length; i++) {
        if (str[i] === str[i + 1]) count++
        else {
            result += count + str[i]
            count = 1
        }
    }
    return result
}

//run length decoding string
let decode = (str) => {
    let result = ""
    let count = ""
    for (let i = 0; i < str.length; i++) {
        if (isNaN(str[i])) {
            result += str[i].repeat(count)
            count = ""
        } else count += str[i]
    }
    return result
}

console.log(encode(tocode))
console.log(decode(encode(tocode)))

